compiler/plc_diagnostics/src/diagnostics.rs

```diff
@@ -199,12 +202,25 @@ impl Diagnostic {
             .with_error_code("E006")
     }
 
-    pub fn invalid_parameter_count(expected: usize, received: usize, location: SourceLocation) -> Diagnostic {
-        Diagnostic::new(
-            format!(
-                "Invalid parameter count. Received {received} parameters while {expected} parameters were expected.",
-            )).with_error_code("E032")
-            .with_location(location)
+    pub fn invalid_argument_count<T>(expected: usize, actual: usize, location: T) -> Diagnostic
+    where
+        T: Into<SourceLocation>,
+    {
+        // Let's be extra fancy here 🕺
+        fn message(value: usize) -> String {
+            match value {
+                1 => format!("{value} argument"),
+                _ => format!("{value} arguments"),
+            }
+        }
+
+        Diagnostic::new(format!(
+            "this POU takes {expected} but {actual} were supplied",
```

Comment:
> this POU takes 0 arguments but 1 argument were supplied

We could be even more fancy here by changing _were_ to _was_ for single arguments :nail_care:

Reply:
Damn, now we are at peak fancy level 🕺

Reply:
Sorry to be the party p**per but you don't need the check if you say something like
"Expected {x} arguments but got {y}."